apps/cms-v2/src/modules/providers/payloadcms/payloadcms-client.ts

```diff
@@ -0,0 +1,170 @@
+      {
+        where: {
+          [context.configuration.productVariantFieldsMapping.variantId]: {
+            equals: context.variant.id,
+          },
+        },
+      },
+      {
+        addQueryPrefix: true,
+      },
+    );
+
+    try {
+      const response = await fetch(
+        this.constructCollectionUrl(context.configuration) + queryString,
+        {
+          method: "DELETE",
+          headers: this.getHeaders(context),
+        },
+      );
+
+      if (response.status >= 400) {
+        throw new Error("Error while deleting product variant");
+      }
+    } catch (e) {
+      this.logger.error(e);
+
+      throw e;
+    }
+  }
+
+  private getHeaders(context: Context) {
+    const headers = new Headers({
+      "Content-Type": "application/json",
+    });
+
+    /**
+     * https://payloadcms.com/docs/authentication/config#api-keys
+     */
+    if (
+      context.configuration.authToken.length > 0 &&
+      context.configuration.authenticatedUserSlug.length > 0
+    ) {
+      headers.append(
+        "Authorization",
+        `${context.configuration.authenticatedUserSlug} API-Key ${context.configuration.authToken}`,
+      );
+    }
+
+    return headers;
+  }
+
+  uploadProductVariant(context: Context) {
+    this.logger.debug("Trying to upload product variant");
+
+    return fetch(this.constructCollectionUrl(context.configuration), {
+      method: "POST",
+      body: JSON.stringify(this.mapVariantToPayloadFields(context)),
+      headers: this.getHeaders(context),
+    })
+      .then((r) => {
+        if (r.status >= 400) {
+          throw new Error(`Error while uploading product variant: ${r.statusText}`);
+        }
+      })
+      .catch((e) => {
+        this.logger.error(e);
+
+        throw e;
+      });
+  }
+
+  async updateProductVariant({ configuration, variant }: Context) {
+    this.logger.debug("Trying to update product variant");
+
+    const queryString = qs.stringify(
+      {
+        where: {
+          [configuration.productVariantFieldsMapping.variantId]: {
+            equals: variant.id,
+          },
+        },
+      },
+      {
+        addQueryPrefix: true,
+      },
+    );
+
+    try {
+      const response = await fetch(this.constructCollectionUrl(configuration) + queryString, {
+        method: "PATCH",
+        body: JSON.stringify(this.mapVariantToPayloadFields({ configuration, variant })),
+        headers: this.getHeaders({ configuration, variant }),
+      });
+
+      if (response.status >= 400) {
+        throw new Error("Error while updating product variant");
+      }
+    } catch (e) {
+      this.logger.error(e);
+
+      throw e;
+    }
+  }
+
+  async upsertProductVariant(context: Context) {
+    this.logger.debug("Trying to upsert product variant");
+
+    try {
+      this.logger.debug("Trying to upload product variant");
+
+      await this.uploadProductVariant(context);
+    } catch (e) {
+      this.logger.debug("Trying to update product variant");
```

Comment:
**Issue[^1][^2]**:
Wrong message, should be an error.

[^1]: 🛑 Blocking the review. 
[^2]: Possible bug or not considered application state.